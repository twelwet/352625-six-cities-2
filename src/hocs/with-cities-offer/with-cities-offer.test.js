import React from "react";
import renderer from "react-test-renderer";
import withCitiesOffer from "./with-cities-offer.js";

const MockComponent = () => {
  return <div>Mock</div>;
};

const MockComponentWrapped = withCitiesOffer(MockComponent);

it(`withCitiesOffer correctly wrap MockComponent`, () => {

  const props = {
    onOfferHover: jest.fn(),
    onOfferLeave: jest.fn(),
    [`is_premium`]: true,
    [`preview_image`]: `/filepath`
  };

  const tree = renderer
    .create(<MockComponentWrapped {...props} />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
