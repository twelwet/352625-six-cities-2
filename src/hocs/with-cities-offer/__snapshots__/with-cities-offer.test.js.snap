// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withCitiesOffer correctly wrap MockComponent 1`] = `
<article
  className="cities__place-card place-card"
  onMouseLeave={[MockFunction]}
  onMouseOver={[MockFunction]}
>
  <div
    className="place-card__mark"
  >
    <span>
      Premium
    </span>
  </div>
  <div
    className="cities__image-wrapper place-card__image-wrapper"
  >
    <a
      href="#"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="/filepath"
        width="260"
      />
    </a>
  </div>
  <div>
    Mock
  </div>
</article>
`;
